tests/lax_scipy_test.py

```diff
@@ -90,30 +94,54 @@ def _GetArgsMaker(self, rng, shapes, dtypes):
-       "rng_factory": jtu.rand_some_inf_and_nan
-                      if jtu.device_under_test() != "cpu"
-                      else jtu.rand_default,
-       "shape": shape, "dtype": dtype,
-       "axis": axis, "keepdims": keepdims}
-      for shape in all_shapes for dtype in float_dtypes
-      for axis in range(-len(shape), len(shape))
-      for keepdims in [False, True]))
+       "shape_group": shape_group, "dtype": dtype,
+       "axis": axis, "keepdims": keepdims,
+       "return_sign": return_sign, "use_b": use_b}
+      for shape_group in compatible_shapes for dtype in float_dtypes
+      for axis in range(-max(len(shape) for shape in shape_group),
+                         max(len(shape) for shape in shape_group))
+      for keepdims in [False, True]
+      for return_sign in [False, True]
+      for use_b in [False, True]))
   @jtu.skip_on_flag("jax_xla_backend", "xrt")
-  def testLogSumExp(self, rng_factory, shape, dtype, axis, keepdims):
-    rng = rng_factory(self.rng())
+  @jtu.ignore_warning(category=RuntimeWarning,
+                      message="invalid value encountered in log.*")
+  def testLogSumExp(self, shape_group, dtype, axis,
+                    keepdims, return_sign, use_b):
+    if jtu.device_under_test() != "cpu":
+      rng = jtu.rand_some_inf_and_nan(self.rng())
+    else:
+      rng = jtu.rand_default(self.rng())
     # TODO(mattjj): test autodiff
-    def scipy_fun(array_to_reduce):
-      return osp_special.logsumexp(array_to_reduce, axis, keepdims=keepdims)
-
-    def lax_fun(array_to_reduce):
-      return lsp_special.logsumexp(array_to_reduce, axis, keepdims=keepdims)
-
-    args_maker = lambda: [rng(shape, dtype)]
-    self._CheckAgainstNumpy(scipy_fun, lax_fun, args_maker)
-    self._CompileAndCheck(lax_fun, args_maker)
+    if use_b:
+      def scipy_fun(array_to_reduce, scale_array):
+        return osp_special.logsumexp(array_to_reduce, axis, keepdims=keepdims,
+                                     return_sign=return_sign, b=scale_array)
+
+      def lax_fun(array_to_reduce, scale_array):
+        return lsp_special.logsumexp(array_to_reduce, axis, keepdims=keepdims,
+                                     return_sign=return_sign, b=scale_array)
+
+      args_makers = [(lambda: [rng(shape1, dtype), rng(shape2, dtype)])
+                     for shape1 in shape_group for shape2 in shape_group]
+    else:
+      def scipy_fun(array_to_reduce):
+        return osp_special.logsumexp(array_to_reduce, axis, keepdims=keepdims,
+                                     return_sign=return_sign)
+
+      def lax_fun(array_to_reduce):
+        return lsp_special.logsumexp(array_to_reduce, axis, keepdims=keepdims,
+                                     return_sign=return_sign)
+
+      args_makers = [(lambda: [rng(shape, dtype)]) for shape in shape_group]
+    for args_maker in args_makers:
+      self._CheckAgainstNumpy(scipy_fun, lax_fun, args_maker)
+      self._CompileAndCheck(lax_fun, args_maker)
```

Comment:
Any loops like this should be in the parametrization rather than an explicit for-loop (as written, if there are two test failures, the second one may never be reported because it is pre-empted by the earlier failure in the loop. See for example how it's done here: https://github.com/google/jax/blob/8f4ba7e679b889ce8b75ef8fa07a1df947d89e52/tests/pmap_test.py#L1260-L1286